### ISSUE TITLE
Add fake data storage for the development environment

### ISSUE THREAD

**Issue by lvh** [RackSec/desdemona#8]
Right now, the only way to do archival is Rackspace Cloud Files (see #2). That's unfortunate: it means that a development environment can't test those components without access to the network, and access to a specific service at that. It would be much preferable if you could just `docker compose up` (#7) and have that do a thing all by itself. Simply making long-term storage unavailable is unacceptable: it is critical to how desdemona operates, so that would make a development environment unrealistic.

There are a few potential ways to resolve this:

1. Abstract away the storage API, provide a simple (in-memory?) alternative to Cloud Files
2. Provide fakes for Cloud Files

Both have interesting trade-offs. Plan 1 is obviously nice because it opens up the possibility of other storage backends. If we go with plan 1, most of the work in plan 2 should still happen, albeit with a potentially lower priority: plan 2 is how you provide high-assurance testing for the Rackspace Cloud Files backend.

Plan 2 has a number of parts:

- [ ] The upstream project [mimic][mimic] needs to learn how to speak OpenStack Swift/Rackspace Cloud Files. Fortunately mimic has a friendly and active development environment, so adding features to it should be fairly painless.
- [ ] The Rackspace Cloud Files jClouds API may or may not be OK with a different identity API URL, which is how Mimic expects to work.

It's not clear if this should be done before or after #2, which is about adding support for Rackspace Cloud Files altogether.

[mimic]: https://github.com/rackerlabs/mimic

**Comment by lvh** [comment]
I am moving this back to the original milestone, because Desdemona is a continuously deployed project, and the required reporting is around feature maturity, not artificial releases.
